homeassistant/components/ping/coordinator.py

```diff
@@ -51,8 +55,28 @@ def __init__(
         )
 
     async def _async_update_data(self) -> PingResult:
-        """Trigger ping check."""
-        await self.ping.async_update()
+        """Update ping data within a timeout duration."""
+
+        if self._update_task is not None:
+            if not self._update_task.done():
+                # don't queue another task
+                return self._latest_ping_result()
+        else:
+            # queue a task for pinging.
+            # this is done so that pings can have a duration longer than SLOW_UPDATE_WARNING,
+            # but this method never executes for longer than SLOW_UPDATE_WARNING (we just
+            # might report stale results in the meanwhile)
+            self._update_task = asyncio.create_task(self.ping.async_update())
```

Comment:
if the task was not None, but was done, we should create a new task on this update, right?